### PR TITLE
GunGame Percentage Meter

### PR DESCRIPTION
**Pull request by connieprice**
# Added a percentage meter to GunGame, which also displays the next weapon!

https://user-images.githubusercontent.com/5340283/149446727-da0e96eb-a77f-4fc2-9ca1-53fdc249da1c.mp4

- New GunGame meter, by default, everything acts precisely like before. It just adds some visual flair.
- Support for varying reward percentages for Kills, Assists and Executions to let server owners mix up their gameplay a bit more.
- Actual assist support, by default providing no points, but the option is now there. ( Feel free to change this default if you feel it deserves one. )
- Replaced the Archer with a primary-enabled melee, finally no more Archer blocking half the screen.
- Gave the pilot melees "allow_as_primary" mods so they can be used as empty slot weapons.

### CODE REVIEW

**Comment by VITALISED** [comment]
is there anything that can be done about the weapon icons being stretched?

**Comment by connieprice** [comment]
Just reiterating what I said on discord, I'd fix stretching if I could, unfortunately that's hidden in the depths of RUI for the time being.

**Review by VITALISED — CHANGES_REQUESTED** [review]
These changes are good, but did you mean to commit your weapon keyvalue changes?

**Comment by VITALISED** [comment]
Oh my bad sorry lmfao

File: Northstar.Custom/mod/scripts/vscripts/gamemodes/sh_gamemode_gg.gnut [hunk 1]
```diff
@@ -30,119 +33,122 @@ void function CreateGamemodeGG()
...
+
+
 	// final/special weapons
 	// charge rifle
 	GunGameWeapon ggChargeRifle = { weapon = "mp_weapon_defender", ... }
 	file.weapons.append( ggChargeRifle )
-	
+
 	// pulse blade
 	GunGameWeapon ggPulseBlade = { weapon = "mp_weapon_grenade_sonar", mods = [ "pas_power_cell", "amped_tacticals" ], offhandSlot = 0 }
```
Comment (abarichello) [inline Northstar.Custom/mod/scripts/vscripts/gamemodes/sh_gamemode_gg.gnut:145]:
```suggestion
	GunGameWeapon ggPulseBlade = { weapon = "mp_weapon_grenade_sonar", mods = [ "pas_power_cell", "amped_tacticals" ], offhandSlot = OFFHAND_SPECIAL }
```

Moving pulse blade to its normal slot

**Pull request merged**